shared/model_helpers.py

```diff
@@ -57,9 +59,18 @@ def load_model_and_optimizer(optimizer, model, model_path):
 
 def get_model_and_optimizer(model, optimizer=None, checkpoint=None):
     if optimizer is None:
-        optimizer = optim.SGD(model.parameters(), lr=0.01, momentum=0)
+        optimizer = optim.SGD(get_trainable_model_parameters(model), lr=0.01, momentum=0)
 
     if checkpoint:
         load_model_and_optimizer(optimizer, model, checkpoint)
 
     return model, optimizer
+
+
+def get_trainable_model_parameters(
+    model: torch.nn.Module,
+) -> Callable[[], Iterator[torch.nn.Parameter]]:
+    def generator():
+        return filter(lambda x: x.requires_grad, model.parameters())
+
+    return generator
```

Comment:
I think the better implementation should be
```python
for param in model.parameters():
    if param.requires_grad:
        yield param
```

In this way, you can just pass `get_trainable_model_parameters(model)` into the functions

Reply:
resolved and added a unit test